### ISSUE TITLE
Customize vim keybindings (something like a vimrc file)?

### ISSUE THREAD

**Issue by atreyagagan** [rstudio/rstudio#7350]
I love the vim keybinds option in RStudio, but I have a set of customized keybindings that I would like to use in RStudio as well. Something like editing your system's vimrc file, but applied to RStudio's vim option. Is there a way to do this?

**Comment by jmcphers** [comment]
No, there's not currently a way to do this. Many editor commands can be rebound (in Tools -> Modify Keyboard Shortcuts), and that generates a file in `~/.config/rstudio/keybindings` that you can examine, but the typical vim rebindings (like binding `jk` to <kbd>Esc</kbd> aren't possible.

**Comment by woodspock** [comment]
Looks like `:imap jj <Esc>` is working now. 
But you have to type it every time you open RStudio.

**Comment by multimeric** [comment]
I'm sure it's possible to customize the keymap. The Ace editor definitely supports it, and we have an internal keymap for it: https://github.com/rstudio/rstudio/blob/4f7063b13bb0bae8377639a36d3ff4b7626de6e3/src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/ace/keybinding-vim-uncompressed.js#L832. I just can't work out how to hook the RStudio settings into the Ace editor settings.

**Comment by bradrn** [comment]
I would also like to comment in favour of this proposal. I just discovered Vim mode in RStudio, and it seems to work really well, but this is annoying.

**Comment by jjcurtin** [comment]
As a regular user of vim in RStudio, I'll add support for this.   I end up typing :imap jk <Esc> every time I start a new session, which gets pretty tedious.   There are other mappings I'd like from my .vimrc but at least the option to implement jk to escape is likely used by most vim users

**Comment by Mickychen00** [comment]
Second this feature.

**Comment by sssammm88** [comment]
I would also very much appreciate this feature. Simple keybindings like imap jk would make the vim experience much more enjoyable.

**Comment by sssammm88** [comment]
Assuming the dev team has the capacity, I would still be very interested in that feature.

**Comment by Achim63** [comment]
I guess adding this would open a whole can of worms (if not Pandora's box).

My "workaround" for the jk=esc "issue": I use caps lock as escape key. In macOS this can be set in System Preferences > Keyboard > Shortcuts > Special keys.

As rstudio reloads a file that was edited outside automatically already, a better solution might be a vim/neovim plugin to get the Cmd/Ctrl+Return (Run) and Source functionality inside of vim, i.e. some way/API to trigger reload and execution of the current line/paragraph/script.
